ethereum/blockcreation/src/main/java/org/hyperledger/besu/ethereum/blockcreation/BlockTransactionSelector.java

```diff
@@ -154,7 +225,7 @@ public TransactionSelectionResults buildTransactionListForBlock() {
    * @return The {@code TransactionSelectionResults} results of transaction evaluation.
    */
   public TransactionSelectionResults evaluateTransactions(final List<Transaction> transactions) {
-    transactions.forEach(this::evaluateTransaction);
+    transactions.forEach(transaction -> evaluateTransaction(transaction, true));
```

Comment:
why true here for `reportFutureNonceTransactionsAsInvalid`

Reply:
In this case is true because the semantics of this method is to create a block with the given list of transactions in the given order. So, if the transaction nonce is not correct it's not possible to include it in the block.